### PR TITLE
Treat master records differently

### PR DESCRIPTION
**Pull request by aroltsch**
Master records for video are stored in the `hearshed` table, however, the column values differ from the regular hearing record. We need to handle master records differently. 
1. If it is a master record, don’t create a hearing in the DB, only in memory
2. If it is a master record (either video or travel_board), set hearing_type and regional_office_key based on certain conditions
3. return `master_record` boolean value as part of the hearing object to the FE

### CODE REVIEW

File: app/models/concerns/regional_office.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module RegionalOffice
```
Comment (sharonwarner) [inline app/models/concerns/regional_office.rb]: 👍

File: lib/generators/hearing.rb [hunk 2]
```diff
@@ -14,7 +14,9 @@ def default_attrs
         contentions: "The veteran believes their knee is hurt",
         evidence: "Medical exam occurred on 10/10/2008",
         military_service: "Veteran was in the Vietnam War",
-        comments_for_attorney: "Look for knee-related medical records"
+        comments_for_attorney: "Look for knee-related medical records",
+        regional_office_key: VACOLS::RegionalOffice::CITIES.keys.sample,
+        master_record: [true, false].sample
```
Comment (sharonwarner) [inline lib/generators/hearing.rb]: Does this make sense? Don't these all get saved to the database? Also, this mocked hearing has fields that a master record won't have.

Comment (aroltsch) [reply lib/generators/hearing.rb]: Ok, so you would like the fakes to be as real as possible. `master_record` or `regional_office_key` fields are not saved in the DB, they are in-memory vars.

Comment (aroltsch) [reply lib/generators/hearing.rb]: In real life, the master record will have all these fields but they will all be nil.

**Pull request merged**